app/main/routes.py

```diff
@@ -1,17 +1,37 @@
-from flask import redirect, render_template, url_for
+from flask import redirect, render_template, request, url_for
 from werkzeug.exceptions import HTTPException
 
+from app.const import MIMETYPE
 from app.main import bp
+from app.main.data_requests import post_ingest
 
 
 @bp.route("/", methods=["GET"])
 def index():
     return redirect(url_for("main.upload"))
 
 
-@bp.route("/upload", methods=["GET"])
+@bp.route("/upload", methods=["GET", "POST"])
 def upload():
-    return render_template("upload.html")
+    if request.method == "GET":
+        return render_template("upload.html")
+
+    if request.method == "POST":
+        excel_file = request.files.get("ingest_spreadsheet")
+        file_format = excel_file.content_type
+        if file_format != MIMETYPE.XLSX:
+            response = [f"Unexpected file format: {file_format}"]
+            return render_template("upload.html", response=response)
+
+        response = post_ingest(excel_file, {"source_type": "tf_round_three"})
```

Comment:
Can we add a TODO. This will presumably needed to be changed to `round_four` when available?